### ISSUE TITLE
EnvironmentSecretsPropertyEnvironmentPostProcessor not working as expected

### ISSUE THREAD

**Issue by Radiergummi** [V1ncNet/secrets-spring-boot#5]
I tried to set up a project to read environment configuration. Using the default combination of all four post processors works, but I cannot get it to accept a `_FILE` variable. 

For a simple compose configuration like this:
```yaml
secrets:
  database-password:
    environment: DATABASE_PASSWORD # assume this is set to "foo"
services:
  app:
    build: .
    environment:
      SPRING_DATASOURCE_PASSWORD_FILE: /run/secrets/password
    secrets:
      - source: database-password
        target: /run/secrets/password
```
and the following `META-INF/spring.factories`:
```ini
org.springframework.boot.env.EnvironmentPostProcessor=\
  de.vinado.boot.secrets.EnvironmentSecretsPropertyEnvironmentPostProcessor
```

with no additional configuration specified, I would have expected to end up with `spring.datasource.password=foo`. It remains unset, however.

For the time being, I resorted to enabling all post processors and rely on implicit files below `/run/secrets`, which works; but actually being able to set `_FILE` variables would be great.

**Comment by V1ncNet** [comment]
I'll take a look at the problem in the next few days and provide a fix or more documentation if necessary.

**Comment by V1ncNet** [comment]
TL:DR: Prefix the secret's location with `file:`. This will work. Adjust your Compose as illustrated:

```diff
    environment:
-      SPRING_DATASOURCE_PASSWORD_FILE: /run/secrets/password
+      SPRING_DATASOURCE_PASSWORD_FILE: file:/run/secrets/password
```

---

Just found the issue. The base class `SecretsEnvironmentPostProcessor` almost exclusively uses Spring's `DefaultResourceLoader` which defaults to `ClassPathResource` when the provided resource location (in your case `/run/secrets/password`) starts with _/_. Investigating this, I found that `ClassPathResource` removes the leading slash from any absolute path. This might actually be a bug within the Spring Framework itself.
Anyway, by making the resource's location a file URI, the resource loader uses another implementation of `Resource` which loads the secret file's content correctly.

I'll adjust my documentation to make this clear.

**Comment by Radiergummi** [comment]
Spitze, vielen Dank!
